### ISSUE TITLE
Request: add support for Emby server provider

### ISSUE THREAD

**Issue by mkavidas** [wako-unofficial-addons/helios#37]
Would be awesome if helios could scrape Emby server. Emby's API should make this pretty easy. I don't think adding it at a provider would work in the current state because of authentication.

**Comment by davidreedmichigan** [comment]
I think it will be better to create a new add-on for this

**Comment by mkavidas** [comment]
issue #23 would make this possible. Don't think this should be closed because I still think that app wide Emby integration would be great because the alternative (making stream providers possible) would require users to create their own provider for their Emby server. Also integrating in the app would allow users to use emby servers even if they don't have the API key.

Emby API documentation [here ](https://github.com/MediaBrowser/Emby/wiki).

In defense on why I would rather not create a new add-on for this is because the idea is to be able to search for content across sources (torrent, emby and hosters) and return the best result. For an implementation of what this looks like (on kodi) check out the Gaia addon. If I was to create a separate addon the user experience would be search for torrents OR emby OR hosters instead of search everything and give me back the best result. With that and improved filtering options (sorting by type,provider,size etc.) this could make helios an all in one experience that would be really cool.

Curious to hear the developers thoughts on this.

**Comment by davidreedmichigan** [comment]
I'm gonna answer here for this issue, #23 and #38. 
First of all, I won't accept code that calls/scrapes non legit API/websites. Helios is only a tool, you use it with your own providers as your own risks. For this, Emby integration is ok.
I'd prefer if Emby could be handled via the provider system, but I don't know how Emby works to know if this is possible and I'm too busy right know to have a look.
If you want to add it directly in Helios, then you could do like the integration for PM or RD.
Basically an interface to enter user's credentials. Some queries to retrieve the data from Emby's API that will create an array of StreamLinkSource (or another custom class that extends from BaseSource).

As I said I'm too busy right know to do it, so help is appreciated to improve Helios :)
